File: day1/day1.clj
(require '[clojure.string :as str])

(def data
  (map #(read-string %)
    (-> (slurp "day1/day1-input.txt")
        (str/split #"\n"))))

(defn count-depth
  [nums]
  (reduce (fn [count [prev next]]
            (cond (< prev next) (inc count)
                  :else count))
    0
    (partition 2 1 nums)))


(def windows
  (->> (partition 3 1 data)
       (map #(reduce + %))))

(print "Part 1:" (count-depth data))
(print "Part 2:" (count-depth windows))

File: day1/day1.rb
file = File.open("day1-test-input.txt")
data = file.read.split("\n").map { |x| x.to_i }


counter = 0
data.each_with_index do |num, index|
  next if index + 1 == data.length

  nextnum = data[index+1]

  if num < nextnum
    counter += 1 
  end
end

puts(counter)
